### PR TITLE
[DOC-526][feature]: added nexus module

### PR DESCRIPTION
**Pull request by ddurnev**
Contains nexus data source for variables

### CODE REVIEW

File: nexus/src/main/scala/com/griddynamics/genesis/nexus/datasource/NexusDataSource.scala [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        val encoded = Base64.encodeBase64String((user + ":" + pass).getBytes)
+        con.setRequestProperty("Authorization", "Basic " + encoded)
+        case _ => // no basic auth
+    }
+
+    def getData = {
+        val con  = url.openConnection
+        con.addRequestProperty("accept", "application/json")
+        basicAuth(con)
+        con.connect
```
Comment (rsvato) [inline nexus/src/main/scala/com/griddynamics/genesis/nexus/datasource/NexusDataSource.scala]: may be you should disconnect as well? You can have a lot of keepalive without it. Or set Connection: close header at least

File: nexus/src/main/scala/com/griddynamics/genesis/nexus/datasource/NexusDataSource.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    private def basicAuth(con : URLConnection) = (usernameOpt, passwordOpt) match {
+        case (Some(user), Some(pass)) =>
+        val encoded = Base64.encodeBase64String((user + ":" + pass).getBytes)
+        con.setRequestProperty("Authorization", "Basic " + encoded)
+        case _ => // no basic auth
+    }
+
+    def getData = {
+        val con  = url.openConnection
+        con.addRequestProperty("accept", "application/json")
```
Comment (rsvato) [inline nexus/src/main/scala/com/griddynamics/genesis/nexus/datasource/NexusDataSource.scala]: con.addRequestProperty("Accept", "application/json")

Comment (ddurnev) [inline nexus/src/main/scala/com/griddynamics/genesis/nexus/datasource/NexusDataSource.scala]: "accept" also works

Comment (rsvato) [inline nexus/src/main/scala/com/griddynamics/genesis/nexus/datasource/NexusDataSource.scala]: But common convention is to start all headers from capital letter.

**Comment by ddurnev** [comment]
Fixed.

**Pull request merged**